vncviewer/CConn.cxx

```diff
@@ -373,6 +393,22 @@ void CConn::framebufferUpdateEnd()
 
   updateCount++;
 
+#ifdef WIN32
+  // If Win32 audio output is available, and server has indicated QEMU audio support, and Win32 audio
+  // output is not open yet, then try to open it, set audio format and enable audio streaming on server
+  if (win32AudioOutput && server.supportsQEMUAudio && (!(win32AudioOutput->isOpened()))) {
+    if (win32AudioOutput->openAndAllocateBuffer()) {
+      writer()->writeQemuAudioSetFormat(win32AudioOutput->getSampleFormat(), 
+                                        win32AudioOutput->getNumberOfChannels(),
+                                        win32AudioOutput->getSamplingFreq());
+      writer()->writeQemuAudioEnableOrDisable(true /* enable */);
+    } else {
+      delete win32AudioOutput;
+      win32AudioOutput = NULL;
+    }
+  }
+#endif
```

Comment:
Can this be pushed to `CConnection` instead? Starting the audio on first detection of QEMU audio support should be generic, and ideally handled there.

Reply:
Yes, I think it's possible. Testing of win32AudioOutput against NULL and invokation of win32AudioOutput->openAndAllocateBuffer() unless already opened should remain in CConn, probably in the new virtual method, and the rest of this logic (including calling methods of CMsgWriter) is generic and can be moved up to the base class.